python/setup.py

```diff
@@ -560,13 +560,18 @@ def build(build_python, build_java, build_cpp):
             FutureWarning,
         )
 
-    if not is_automated_build:
-        bazel_precmd_flags = []
     if is_automated_build:
-        root_dir = os.path.join(
-            os.path.abspath(os.environ["SRC_DIR"]), "..", "bazel-root"
-        )
-        out_dir = os.path.join(os.path.abspath(os.environ["SRC_DIR"]), "..", "b-o")
+        if is_native_windows_or_msys():
+            root_dir = os.path.join(
+                os.path.splitdrive(os.environ["SRC_DIR"])[0], "bazel-root"
```

Comment:
could you document an example of `os.environ["SRC_DIR"]` here?

also maybe give `os.path.splitdrive(os.environ["SRC_DIR"])[0]` a variable name? and it won't be calculated twice?

Reply:
> could you document an example of os.environ["SRC_DIR"] here?

It is documented in the [developer docs](https://docs.ray.io/en/latest/ray-contribute/development.html#environment-variables-that-influence-builds)

> give os.path.splitdrive(os.environ["SRC_DIR"])[0]
OK

Reply:
Hmm. The documentation says
> `SRC_DIR`: Can be set to the root of the source checkout, defaults to `None` which is `cwd()`

but I don't see where this magic happens. In the conda-forge build it [is documented](https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-set-during-the-build-process) as being set, but I think in general defining `AUTOMATED_BUILD` and then not defining `SRC_DIR` will error (even before this PR). I will fix that here.

Reply:
Updated to relate to review comments